modules/hcp-kube-api-server-cert.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// * hosted-control-planes/hcp-certificates.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-kube-api-server-cert_{context}"]
+= Configuring a custom DNS domain for a hosted cluster's Kubernetes API server
+
+You can configure a custom DNS domain for your hosted cluster's Kubernetes API server and consume the generated `kubeconfig` secret that points to the custom domain.
+
+.Prerequisites
+
+* You have a running hosted cluster.
+
+* You have access to modify the `HostedCluster` resource.
+
+* You have a custom DNS domain to use for the Kubernetes API server.
++
+** The custom DNS domain must be properly configured and resolvable.
+** The DNS domain must have valid TLS certificates configured.
+** Network access to the domain must be properly configured in your environment.
+** The custom DNS domain must be unique across your hosted clusters.
+
+* You have a configured custom certificate. For more information, see "Configuring a custom API server certificate in a hosted cluster".
+
+.Procedure
+
+. In your provider platform, configure the DNS record. The DNS record must be properly configured and resolvable from your cluster.
+
+. In your `HostedCluster` specification, modify the `kubeAPIServerDNSName` field, as shown in the following example:
++
+[source,yaml]
+----
+apiVersion: hypershift.openshift.io/v1beta1
+kind: HostedCluster
+metadata:
+  name: <hosted_cluster_name>
+  namespace: <hosted_cluster_namespace>
+spec:
+  configuration:
+    apiServer:
+      servingCerts:
+        namedCertificates:
+        - names: <1>
+          - api-custom-cert-sample-hosted.sample-hosted.example.com
+          servingCertificate: <2>
+            name: sample-hosted-kas-custom-cert
+  kubeAPIServerDNSName: api-custom-cert-sample-hosted.sample-hosted.example.com <3>
+...
+----
+<1> The list of DNS names that the certificate is valid for.
+<2> The name of the secret that contains the custom certificate.
+<3> This field accepts a URI that will be used as the API server endpoint.
+
+. Apply the configuration by entering the following command:
++
+[source,terminal]
+----
+$ oc -f <hosted_cluster_spec>.yaml
+----
++
+After the configuration is applied, the HyperShift Operator generates a new `kubeconfig` secret that points to your custom DNS domain.
+
+. Retrieve the `kubeconfig` secret by entering the following command:
++
+[source,terminal]
+----
+$ kubectl get secret <hosted_cluster_name>-custom-admin-kubeconfig \
+  -n <cluster-namespace> \
+  -o jsonpath='{.data.kubeconfig}' | base64 -d
+----
```

Comment:
I would add a note here. The new kubconfig could be grabbed via:
- shell: as you mentioned above
- MCE/ACM Console: Once we have created that custom-kubeconfig, just go to the HostedCluster under the MCE console and press the "download kubeconfig" button.

You cannot consume this new kubeconfig via "show login command" in the Openshift console.